### DISCUSSION TITLE
C# scripts run differently in TE2 & TE3

### DISCUSSION THREAD

**Discussion by m-kovalsky** [TabularEditor/TabularEditor3#1144] [🙋♀️TE3 Q&A] [answered]
There is a scaling issue when I run the MiniModelBuilder tool in TE3. This doesn't happen in TE2. Is there an inherent difference in running c# scripts between the 2 programs?

https://github.com/m-kovalsky/MiniModelBuilder/blob/main/MiniModelBuilder.cs

**Comment by otykier** [comment]
Could you post a screenshot to illustrate the issue? In general, the scripts execute in the same way, but the problem is that TE3 has high-DPI enabled at the application level, where as TE2 doesn't, so any UI you bring up through a C# script will inherit the DPI settings of the application.

**Comment by m-kovalsky** [comment]
Here's a screenshot of the script executed in TE2:
![image](https://user-images.githubusercontent.com/29556918/168883320-6537a074-ec82-455a-a777-fdc6a0cb3609.png)

Here's a screenshot of the same script executed in TE3 (with the System.Xml assembly imported in the script):
![image](https://user-images.githubusercontent.com/29556918/168883519-e741336b-4176-4c26-891d-d7457242bfd8.png)

**Comment by otykier** [comment, marked as answer]
Based on [this post](https://stackoverflow.com/questions/4009150/c-sharp-winforms-disable-dpi-scaling) it may be possible to fix the scaling in TE3 by setting the `AutoScaleMode` property and manually changing the font size:

```csharp
var scaledFont = new Font(newForm.Font.Name, 8.25f * 96f / newForm.CreateGraphics().DpiX, newForm.Font.Style, newForm.Font.Unit, newForm.Font.GdiCharSet, newForm.Font.GdiVerticalFont);
newForm.Font = scaledFont;
newForm.AutoScaleMode = System.Windows.Forms.AutoScaleMode.None;
```
